followthemoney/types/registry.py

```diff
@@ -8,43 +10,40 @@ class Registry(object):
     then clean, validate or format values of that type."""
 
     def __init__(self):
-        self._types = {}
+        self._types: Dict[str, PropertyType] = {}
 
-    def add(self, clazz):
+    def add(self, clazz: type):
         if issubclass(clazz, PropertyType):
             self._types[clazz.name] = clazz()
 
     @property
-    def types(self):
+    def types(self) -> List[PropertyType]:
         """Return all types known to the system."""
         return list(self._types.values())
 
     @property
-    def matchable(self):
+    def matchable(self) -> List[PropertyType]:
         """Return all matchable property types."""
         return [t for t in self.types if t.matchable]
 
     @property
-    def pivots(self):
+    def pivots(self) -> List[PropertyType]:
         return [t for t in self.types if t.pivot]
 
     @property
-    def groups(self):
+    def groups(self) -> Dict[str, PropertyType]:
         return {t.group: t for t in self.types if t.group is not None}
 
-    def get(self, name):
+    def get(self, name: Union[str, PropertyType]) -> PropertyType:
         """For a given property type name, get its handling object."""
         # Allow transparent re-checking.
         if isinstance(name, PropertyType):
             return name
-        return self._types.get(name)
+        return self._types[name]
```

Comment:
This is behaviour change. Can this be a separate PR?

Reply:
The reason I changed it is because using `get` turns the return type to `Optional[PropertyType]`... and everything that calls `get` or `registry[<name>]` can be `None`.

So I can change that, but that will also mean a lot of type checking down the line:

```
$ (git::typing) mypy . 
followthemoney/types/registry.py:46: error: List comprehension has incompatible type List[Optional[PropertyType]]; expected List[PropertyType]
followthemoney/schema.py:117: error: Item "None" of "Optional[PropertyType]" has no attribute "name"
followthemoney/proxy.py:186: error: Argument 1 to "get_type_values" of "EntityProxy" has incompatible type "Optional[PropertyType]"; expected "PropertyType"
followthemoney/proxy.py:202: error: Item "None" of "Optional[PropertyType]" has no attribute "rdf"
followthemoney/proxy.py:224: error: Argument 1 to "get_type_values" of "EntityProxy" has incompatible type "Optional[PropertyType]"; expected "PropertyType"
followthemoney/proxy.py:228: error: Argument 1 to "get_type_values" of "EntityProxy" has incompatible type "Optional[PropertyType]"; expected "PropertyType"
followthemoney/property.py:32: error: Incompatible types in assignment (expression has type "Optional[PropertyType]", variable has type "PropertyType")
followthemoney/namespace.py:24: error: Item "None" of "Optional[PropertyType]" has no attribute "clean"
followthemoney/graph.py:55: error: Argument 1 to "Node" has incompatible type "Optional[PropertyType]"; expected "PropertyType"
followthemoney/compare.py:20: error: Dict entry 0 has incompatible type "Optional[PropertyType]": "int"; expected "PropertyType": "float"
followthemoney/compare.py:21: error: Dict entry 1 has incompatible type "Optional[PropertyType]": "int"; expected "PropertyType": "float"
followthemoney/compare.py:22: error: Dict entry 2 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:23: error: Dict entry 3 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:24: error: Dict entry 4 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:25: error: Dict entry 5 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:26: error: Dict entry 6 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:27: error: Dict entry 7 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:28: error: Dict entry 8 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:29: error: Dict entry 9 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/compare.py:30: error: Dict entry 10 has incompatible type "Optional[PropertyType]": "float"; expected "PropertyType": "float"
followthemoney/export/graph.py:9: error: Item "None" of "Optional[PropertyType]" has no attribute "name"
Found 21 errors in 8 files (checked 101 source files)
```